examples/src/main/java/com/mapbox/navigation/examples/core/MapboxNavigationActivity.kt

```diff
@@ -494,6 +545,19 @@ class MapboxNavigationActivity : AppCompatActivity() {
         )
     }
 
+    private fun setRoutesPreview(routes: List<NavigationRoute>) {
+        binding.navigateButton.apply {
+            visibility = View.VISIBLE
+            setOnClickListener {
+                visibility = View.GONE
+                setRouteAndStartNavigation(mapboxNavigation.getRoutesPreview()!!.routesList)
+                mapboxNavigation.setRoutesPreview(emptyList())
```

Comment:
is it necessary to remove it at all? seems like it's ok to call setNavigationRoute and be done

Reply:
It's not necessary for the SDK, but it is necessary for this particular example. It registers `previewMapClickListener` if route preview is active and unregisters it when routes preview are cleaned up.

Reply:
> Because it looks like most of the time the developer would want to both end preview state and enter AG.

Yes, but it appeared that route preview should also cover a case to preview new route in parallel to current navigation session.